### ISSUE TITLE
Poor performance of async functions

### ISSUE THREAD

**Issue by jakubneubauer** [microsoft/ClearScript#356]
Hello,
I compared running a simple pure-javascript program with ClearScript V8 vs. NodeJS. The program does nothing special, just pure javascript, without calls to .net CLR etc.
When running synchronous code, the results are perfectly comparable, almost the same.
However, using asynchronous functions ClearScript runs cca 9x slower.
I think that some reasonable "slowness factor" is acceptable, but 9-times, that is too much I believe.

C# side is quiet simple, using engine with defaults:
```
var script = LoadScript("bench2.js");
var engine = new V8ScriptEngine();
engine.Execute(script);
```

NodeJS using from commandline:
```
nodejs bench2.js
```

I attached the 2 .js files - one "synchronous" version and the second one "async" version.

Am I doing something wrong? Is there any option/feature that affects async handling?

- nodejs v16.13.2
- ClearScript  7.2.2
- .net 6.0.101

PS: Github didn't allow attach .js files, so I used extension .txt

**Comment by jakubneubauer** [comment]
[bench2.txt](https://github.com/microsoft/ClearScript/files/8122553/bench2.txt)
[bench2-async.txt](https://github.com/microsoft/ClearScript/files/8122554/bench2-async.txt)

**Comment by ClearScriptLib** [comment]
Hi @jakubneubauer,

Thanks for reporting this! In our testing, the difference is around 4x rather than 9x, but that may have to do with hardware details such as CPU core count.

In any case, we'll have a fix in the next release. Thanks again!

**Comment by ClearScriptLib** [comment]
Fixed in [Version 7.2.3](https://github.com/microsoft/ClearScript/releases/tag/7.2.3).
